### PR TITLE
Update EIP-2771: Add myself as an author and bring it up to date

### PR DESCRIPTION
**Pull request by Pandapip1**
CC other authors: Ronan Sandford (@wighawag), Liraz Siri (@lirazsiri), Dror Tirosh (@drortirosh), Yoav Weiss (@yoavw), Alex Forshtat (@forshtat), Hadrien Croubois (@Amxx), Sachin Tomar (@tomarsachin2271), Patrick McCorry (@stonecoldpat), Nicolas Venturo (@nventuro), Fabian Vogelsteller (@frozeman)

### CODE REVIEW

**Comment by Amxx** [comment]
I'd love to have the cat herder's opinion on that.

I remember taking care of EIP-1967 finalization (including late updates and the PEEPanEIP call) without being added as an author.

**Comment by Amxx** [comment]
[EIP-1](https://eips.ethereum.org/EIPS/eip-1) says:

>  Transferring EIP Ownership
>
>It occasionally becomes necessary to transfer ownership of EIPs to a new champion. In general, we’d like to retain the original author as a co-author of the transferred EIP, but that’s really up to the original author. A good reason to transfer ownership is because the original author no longer has the time or interest in updating it or following through with the EIP process, or has fallen off the face of the ‘net (i.e. is unreachable or isn’t responding to email). A bad reason to transfer ownership is because you don’t agree with the direction of the EIP. We try to build consensus around an EIP, but if that’s not possible, you can always submit a competing EIP.
>
> If you are interested in assuming ownership of an EIP, send a message asking to take over, addressed to both the original author and the EIP editor. If the original author doesn’t respond to the email in a timely manner, the EIP editor will make a unilateral decision (it’s not like such decisions can’t be reversed :)).

My interpretation of that (other co-authors may disagree) is that championing the EIP doesn't make you an author unless you actually participate in the EIP content (by fixing issues, adding details, ...)

**Comment by Pandapip1** [comment]
> My interpretation of that (other co-authors may disagree) is that championing the EIP doesn't make you an author unless you actually participate in the EIP content (by fixing issues, adding details, ...)

I do believe that my contributions (see the [files tab](https://github.com/ethereum/EIPs/pull/5607/files)) are sufficient for this purpose. In addition, championing an EIP generally has been considered an acceptable condition for coauthorship.

**Comment by Pandapip1** [comment]
> I remember taking care of EIP-1967 finalization (including late updates and the PEEPanEIP call) without being added as an author.

Feel free to retroactively add yourself as an author - I definitely think that qualifies you for it!

**Comment by Pandapip1** [comment]
> Note: no need to worry about 1967.

Too late, you've already been added as a co-author :)

**Comment by Pandapip1** [comment]
@Amxx - any more updates?

**Comment by forshtat** [comment]
> > My interpretation of that (other co-authors may disagree) is that championing the EIP doesn't make you an author unless you actually participate in the EIP content (by fixing issues, adding details, ...)
> 
> I do believe that my contributions (see the [files tab](https://github.com/ethereum/EIPs/pull/5607/files)) are sufficient for this purpose. In addition, championing an EIP generally has been considered an acceptable condition for coauthorship.

Hello @Pandapip1 

Thanks for the PR.

May I ask, what was changed in the content of the EIP itself?

The github diff is showing every line has changed, but that seems to be an issue with newlines, whitespaces and lines moved around. This makes it hard to review the changes, if there are any.

**Comment by Pandapip1** [comment]
Mostly, content was moved around, and some sections were rewritten:

- The simple summary section was removed and shortened to fit in the description preamble entry
- The newlines from all sections were stripped, and there was some minor rephrasing throughout (nothing that changed the actual content)
- The reference implementation was removed, as it is contained in an external repository (which is no longer allowed). This should be re-added later.
- The example flow in the specification was moved into its own subsection
- The definitions were moved into their own subsection
- `isTrustedForwarder` was changed to `view` (this is an actual change to the spec!)
- The behavior of `isTrustedForwarder` was not defined. A definition was added.
- Removed the 50k gas limit, as EIPs cannot put fixed gas limits in place because gas costs can change
- Moved the rationale section (it was in the wrong order)
- Moved the example recipient from the specification to the reference implementation

**Comment by Pandapip1** [comment]
The conditions for authorship are also a lot less strict for EIPs than for academia in general. Helping to move an EIP through the EIP process and making minor changes (which is what this EIP does) does warrant co-authorship (see https://github.com/ethereum/EIPs/pull/5621). Since this EIP has been stagnant for >6 months, the "following through with the EIP process" portion of transferring authorship pretty uncontroversially applies here.

**Review by Vladisvob — APPROVED** [review]

**Comment by xinbenlv** [comment]
Here is my general views of _this type of request_:
1. EIP adoption shall have existing author approval
2. The title "author" IMHO are reserved for a role that contribute to the content of a work. Purely editorial change, e.g. styling or trimming, doesn't by default be recognized as author, unless recognized by existing authors. 

For _this particular request_, here is my view:
1. I am in full support of reviving this EIP. 
2. For whether to add @Pandapip1  author, I'd like to respect existing authors' decision.
3. Existing authors shall also note that the content of EIP-2771 _was_ published under CC0 which means if this EIP lacks activity, @Pandapip1  or any other person shall be entitled to a freedom of starting a competing EIP, IMHO even with the exact same content, except that the competing EIP will have a new EIP number.

**Comment by Pandapip1** [comment]
> IMHO even with the exact same content

Actually, legally with the exact same content.

I would probably make a few tweaks, primarily trimming and rewording some of the non-normative sections. Most of those changes are in this PR.

File: EIPS/eip-2771.md
Comment (xinbenlv) [inline EIPS/eip-2771.md:5]: Defer to authors to decide.

Comment (wighawag) [reply EIPS/eip-2771.md:5]: personally I am fine with this change. I am glad someone took the lead to improve the wordings with the goal to make the EIP final

Comment (Pandapip1) [reply EIPS/eip-2771.md:5]: @wighawag would you mind approving this PR? https://github.com/ethereum/EIPs/pull/5607/files#submit-review

File: EIPS/eip-2771.md
Comment (wighawag) [inline EIPS/eip-2771.md:15]: Getter function is an implementation details, we should mention that the msg sender is encoded at the end of the calldata

**Review by wighawag — CHANGES_REQUESTED** [review]

**Review by wighawag — APPROVED** [review]

**Pull request merged**
